subsys/bluetooth/host/conn.c

```diff
@@ -2338,6 +2338,24 @@ void bt_conn_cb_register(struct bt_conn_cb *cb)
 	callback_list = cb;
 }
 
+void bt_conn_cb_unregister(struct bt_conn_cb *cb)
+{
+	struct bt_conn_cb *previous_callback = callback_list;
+
+	if (callback_list == cb) {
+		callback_list = callback_list->_next;
+		return;
+	}
+
+	while (previous_callback->_next) {
+		if (previous_callback->_next == cb) {
+			previous_callback->_next = previous_callback->_next->_next;
+			return;
+		}
+		previous_callback = previous_callback->_next;
+	}
```

Comment:
If `cb` is never found in the callback list, should we return an error?

Reply:
HI Thalley, 
I think it's not need to return an error.
1. this is use to remove a callback, not to find a callback, if the callback not register before, it's ok.
2. It is better to have a uniform function format, like bt_le_scan_cb_unregister has no return.
thanks.

Reply:
Hmm, I think `bt_le_scan_cb_unregister` might be an outlier here, as `bt_conn_auth_info_cb_unregister`, `bt_gatt_service_unregister`, `bt_iso_server_unregister`, `bt_has_preset_unregister`, `bt_pacs_cap_unregister` and `bt_ots_client_unregister` all return an error if the thing being unregistered was not registered first. 

I think it would make sense to return -ENOENT or -EALREADY if the provided call is currently not registered. @jhedberg what do you think?

Reply:
HI @Thalley @jhedberg , I submit the code again, please help to review, thanks.

Reply:
Hi @Thalley @jhedberg , I submit the code again, please help to review, thanks.